### PR TITLE
Jha eligibility

### PR DESCRIPTION
**Pull request by Ian-c-cameron**
### Please check if the PR fulfills these requirements:

- [x] After these changes, the app was run and still works as expected
- [ ] Unit tests for these changes were added (if applicable)
- [X] All existing unit tests were run and still pass

### Please specify the type of change this PR introduces (Bug fix, feature addition, content update, chore, etc.):
Add eligibility questionnaires for MSP, FPC, and SB.

### CODE REVIEW

File: jha/src/store/modules/enrolment-module.js [hunk 1]
```diff
@@ -10,6 +10,15 @@ export const SET_CAPTCHA_TOKEN = 'setCaptchaToken';
...
 export const SET_IS_INFO_COLLECTION_NOTICE_OPEN = 'setIsInfoCollectionNoticeOpen';
+// Eligibility Questionnaires
+export const SET_LIVE_IN_BC = 'setLiveInBC';
+export const SET_AWAY_OVER_30 = 'setAwayOver30';
+export const SET_STUDENT_MINOR_REFUGEE = 'setStudentMinorRefugee';
+export const SET_HAS_DOCUMENTS = 'setHasDocuments';
+export const SET_MEETS_FPC_CRITERIA = 'setMeetsFPCCriteria';
+export const SET_HAS_FPC_INFO = 'setHasFPCInfo';
+export const SET_MEETS_SB_CRITERIA = 'setMeetsSBCriteria';
+export const SET_HAS_SB_INFO = 'setHasSBInfo';
```
Comment (harrymaynard-maximus) [inline jha/src/store/modules/enrolment-module.js:21]: Can you prefix these with "EQ" (eligibility questionnaire)? It makes it easier to identify the usage. For example:
```javascript
export const SET_EQ_LIVE_IN_BC = 'setEQLiveInBC';
export const SET_EQ_AWAY_OVER_30 = 'setEQAwayOver30';
```
Same with the state variables in this module.

File: jha/src/views/enrolment/FPCareEligibilityPage.vue [hunk 2]
```diff
@@ -2,12 +2,57 @@
...
+        <p class="mt-0">Answering the following questions will help you determine which programs you are eligible for, and ensure you have the required documents and information to complete your application(s).</p>
         <hr class="mt-0"/>
-        <h2>Fair PharmaCare</h2>
+        <h2>Fair PharmaCare Enrolment</h2>
+        <Radio
+          id='apply-fpc'
+          name='apply-fpc'
+          label='1. Will you use the Joint Health Application to register for Fair PharmaCare?'
+          v-model='applyFPC'
+          :items='radioOptionsApplyFPC' />
```
Comment (harrymaynard-maximus) [inline jha/src/views/enrolment/FPCareEligibilityPage.vue:15]: The `Radio` usages are missing `blur` handlers. They need to validate the fields upon tabbing out.

File: jha/src/views/enrolment/FPCareEligibilityPage.vue [hunk 3]
```diff
@@ -50,24 +120,55 @@ export default {
...
+    const validations = {
+      applyFPC: {
+        validateQuestionsAnswered,
+      }
+    };
     return validations;
   },
   methods: {
     validateFields() {
+      this.saveData();
```
Comment (harrymaynard-maximus) [inline jha/src/views/enrolment/FPCareEligibilityPage.vue:159]: I'm think we want to save the data to the Vuex store only if the validation passes. So this should be after the validation check.

**Review by harrymaynard-maximus — CHANGES_REQUESTED** [review]
A few comments. They apply to all 3 pages.

File: jha/src/views/enrolment/FPCareEligibilityPage.vue (hunk 2 above)
Comment (Ian-c-cameron) [reply jha/src/views/enrolment/FPCareEligibilityPage.vue:15]: They don't need to validate at all... the only validation required for the questionnaires is to stop the user being able to continue if they haven't answered the last question asked.  That error should only be presented if they click continue.  Therefore blur handler is not needed.

File: jha/src/views/enrolment/FPCareEligibilityPage.vue (hunk 3 above)
Comment (Ian-c-cameron) [reply jha/src/views/enrolment/FPCareEligibilityPage.vue]: fixed... this was copied from Spouse info, so it should be changed there.

File: jha/src/store/modules/enrolment-module.js (hunk 1 above)
Comment (Ian-c-cameron) [reply jha/src/store/modules/enrolment-module.js]: Changed variable names, also updated names to move identifier for which eligibility page the variable goes with.

**Review by harrymaynard-maximus — APPROVED** [review]
Looks good. Nice work!

**Pull request merged**